### PR TITLE
4 loads a new level

### PR DESCRIPTION
**Pull request by flimflamjim**
Still need to sort out edge cases of RegExp matches on ends of strings. That bloody next level button is killing me.

### CODE REVIEW

File: test/e2e/app.spec.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ describe("regexpert", function(){
   beforeEach(function(){
     browser.get('/');
     levelText = 'Horse brHeeding is reproduction in horses, and particularly the human-directed process of selective breeding of animals, particularly purebred horses of a given breed. Planned matings can be used to produce specifically desired characteristics in domesticated horses. Furthermore, modern breeding management and technologies can increase the rate of conception, a healthy pregnancy, and successful foaling.';
```
Comment (gerauf) [inline test/e2e/app.spec.js]: these tests are all really brittle - can't change content without all tests failing - needs refactoring

File: test/e2e/app.spec.js [hunk 2]
```diff
@@ -17,8 +19,30 @@ describe("regexpert", function(){
 
   it('a game can be won', function(){
     expect($('span#win-message').getText()).not.toMatch('WINNER');
-    expect($('p#level-text').getText()).toEqual(levelText);
     $('input#user-input').sendKeys('\\b[a-z]..\\b');
     expect($('span#win-message').getText()).toEqual('WINNER');
   });
+
+  it('a next level button appears when the game is won', function(){
```
Comment (gerauf) [inline test/e2e/app.spec.js]: probably not needed for a feature test - they add a lot of overhead to run time so we should only be testing whats absolutely necessary - maybe move expectation into another test which is already looking for win conditions

File: test/e2e/app.spec.js [hunk 3]
```diff
@@ -17,8 +19,30 @@ describe("regexpert", function(){
...
+    $('input#user-input').sendKeys("\\b[a-z]..\\b");
+    expect($('button#next-level').isPresent()).toBe(true);
+  });
+
+  it('starts on level 1', function() {
+    expect($('p#level-text').getText()).toEqual(levelText);
+  });
+
+  it('loads level 2', function() {
+    $('input#user-input').sendKeys("\\b[a-z]..\\b");
```
Comment (gerauf) [inline test/e2e/app.spec.js]: same as above

File: test/e2e/app.spec.js [hunk 4]
```diff
@@ -17,8 +19,30 @@ describe("regexpert", function(){
...
+    $('button#next-level').click();
+    expect($('section#level-number').getText()).toEqual('Level: 2');
+    expect($('p#level-text').getText()).toEqual(levelText2);
+  });
+
+  it('removes the next level button on click', function() {
+    $('input#user-input').sendKeys("\\b[a-z]..\\b");
```
Comment (gerauf) [inline test/e2e/app.spec.js]: same as above

File: test/unit/levelService.spec.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ describe('LevelService', function(){
 
   beforeEach(module('regexpert'));
 
-  var LevelService, httpBackend, url, response;
+  var LevelService, LevelFactory, httpBackend, url, response;
```
Comment (gerauf) [inline test/unit/levelService.spec.js]: levelFactory doesn't need to be injected into the tests

**Comment by gerauf** [comment]
🐴
